### PR TITLE
Lp 10604 register device

### PR DESCRIPTION
**Pull request by gg4race**

### CODE REVIEW

File: Leanplum/Api/LPRegisterDeviceApi.m [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        NSArray *responseArray = [response valueForKey:@"response"];
+        NSDictionary *resultDict = responseArray[0];
+        if (error != nil) {
+            failure(error);
+        }
+        else {
+            if ([resultDict objectForKey:@"success"]) {
+                success();
+            } else {
+                NSError *error = [LPErrorHelper makeResponseError:@{@"message": @"Invalid Input"}];
```
Comment (hrishileanplum) [inline Leanplum/Api/LPRegisterDeviceApi.m]: We need to revisit this method . makeResponseError. 
The real usecase has to be that we pass the http error to this function and then it creates a response for us. Not, that we set error and message here.

File: LeanplumTests/LPRegisterDeviceApiTests.m [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Error: %@", error);
+        }
+    }];
+}
+
+- (void)testRegisterDeviceApiWithAttributesStub {
+    [LPTestHelper setupStub:200 withFileName:@"simple_success_response.json"];
```
Comment (hrishileanplum) [inline LeanplumTests/LPRegisterDeviceApiTests.m]: one quick note. simple_success_response.json, can we please update the name to . "simple_post_success_response.json"

File: Leanplum/Api/LPRegisterDeviceApi.m (hunk 1 above)
Comment (gg4race) [reply Leanplum/Api/LPRegisterDeviceApi.m]: changed makeUserInfo, now default error message should be attached if no error message is received from server

File: LeanplumTests/LPRegisterDeviceApiTests.m (hunk 2 above)
Comment (gg4race) [reply LeanplumTests/LPRegisterDeviceApiTests.m]: changed for simple_success and simple_error

**Comment by gg4race** [comment]
for Register Device, i checked the only param input in the old SDK is LP_PARAM_EMAIL, which i check for in the new sdk

File: LeanplumTests/LPRegisterDeviceApiTests.m (hunk 2 above)
Comment (hrishileanplum) [reply LeanplumTests/LPRegisterDeviceApiTests.m]: your awesome

**Pull request merged**